pkg/sbom/cyclonedx/testdata/happy/nested-packages-bom.json

```diff
@@ -0,0 +1,175 @@
+          "name": "aquasecurity:trivy:PkgID",
+          "value": "github.com/aquasecurity/go-pep440-version@v0.0.0-20210121094942-22b2f8951d46"
+        },
+        {
+          "name": "aquasecurity:trivy:PkgType",
+          "value": "gobinary"
+        }
+      ]
+    },
+    {
+      "bom-ref": "pkg:golang/github.com/aquasecurity/go-version@v0.0.0-20210121072130-637058cfe492",
+      "type": "library",
+      "name": "github.com/aquasecurity/go-version",
+      "version": "v0.0.0-20210121072130-637058cfe492",
+      "purl": "pkg:golang/github.com/aquasecurity/go-version@v0.0.0-20210121072130-637058cfe492",
+      "properties": [
+        {
+          "name": "aquasecurity:trivy:PkgID",
+          "value": "github.com/aquasecurity/go-version@v0.0.0-20210121072130-637058cfe492"
+        },
+        {
+          "name": "aquasecurity:trivy:PkgType",
+          "value": "gobinary"
+        }
+      ]
+    },
+    {
+      "bom-ref": "pkg:golang/github.com/aquasecurity/test",
+      "type": "library",
+      "name": "github.com/aquasecurity/test",
+      "purl": "pkg:golang/github.com/aquasecurity/test",
+      "properties": [
+        {
+          "name": "aquasecurity:trivy:PkgID",
+          "value": "github.com/aquasecurity/test"
+        },
+        {
+          "name": "aquasecurity:trivy:PkgType",
+          "value": "gobinary"
+        }
+      ]
+    },
+    {
+      "bom-ref": "pkg:golang/golang.org/x/xerrors@v0.0.0-20200804184101-5ec99f83aff1",
+      "type": "library",
+      "name": "golang.org/x/xerrors",
+      "version": "v0.0.0-20200804184101-5ec99f83aff1",
+      "purl": "pkg:golang/golang.org/x/xerrors@v0.0.0-20200804184101-5ec99f83aff1",
+      "properties": [
+        {
+          "name": "aquasecurity:trivy:PkgID",
+          "value": "golang.org/x/xerrors@v0.0.0-20200804184101-5ec99f83aff1"
+        },
+        {
+          "name": "aquasecurity:trivy:PkgType",
+          "value": "gobinary"
+        }
+      ]
+    },
+    {
+      "bom-ref": "pkg:golang/stdlib@v1.15.2",
+      "type": "library",
+      "name": "stdlib",
+      "version": "v1.15.2",
+      "purl": "pkg:golang/stdlib@v1.15.2",
+      "properties": [
+        {
+          "name": "aquasecurity:trivy:PkgID",
+          "value": "stdlib@v1.15.2"
+        },
+        {
+          "name": "aquasecurity:trivy:PkgType",
+          "value": "gobinary"
+        }
+      ]
+    }
+  ],
+  "dependencies": [
+    {
+      "ref": "1cb40520-a22c-481f-ad77-6bc6960430c5",
+      "dependsOn": [
+        "4021d631-e242-4e69-8a93-928665810a27"
+      ]
+    },
+    {
+      "ref": "4021d631-e242-4e69-8a93-928665810a27",
+      "dependsOn": [
+        "pkg:golang/github.com/aquasecurity/go-pep440-version@v0.0.0-20210121094942-22b2f8951d46",
+        "pkg:golang/github.com/aquasecurity/go-version@v0.0.0-20210121072130-637058cfe492",
+        "pkg:golang/github.com/aquasecurity/test",
+        "pkg:golang/golang.org/x/xerrors@v0.0.0-20200804184101-5ec99f83aff1"
+      ]
```

Comment:
These dependencies will be changed after #8104

So this should be changed in one of these PRs

Reply:
updated in https://github.com/aquasecurity/trivy/pull/8144/commits/40d5dac6f0891aea3acfad955e5059e2e14ac28d